core/templates/domain/skill/concept-card-backend-api.service.ts

```diff
@@ -17,92 +17,123 @@
  * about the concept card of a skill from the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/skill/skill-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
 
-angular.module('oppia').factory('ConceptCardBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'CONCEPT_CARD_DATA_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService, CONCEPT_CARD_DATA_URL_TEMPLATE) {
-    // Maps previously loaded concept cards to their IDs.
-    var _conceptCardCache = [];
+import cloneDeep from 'lodash/cloneDeep';
 
-    var _fetchConceptCards = function(
-        skillIds, successCallback, errorCallback) {
-      var conceptCardDataUrl = UrlInterpolationService.interpolateUrl(
-        CONCEPT_CARD_DATA_URL_TEMPLATE, {
-          comma_separated_skill_ids: skillIds.join(',')
-        });
+import { ConceptCardObjectFactory } from
+  'domain/skill/ConceptCardObjectFactory';
+import { RecordedVoiceoversObjectFactory } from
+  'domain/exploration/RecordedVoiceoversObjectFactory';
+import { SkillDomainConstants } from
+  'domain/skill/skill-domain.constants';
+import { SubtitledHtmlObjectFactory } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { VoiceoverObjectFactory } from
+  'domain/exploration/VoiceoverObjectFactory';
+import { WorkedExampleObjectFactory } from
+  'domain/skill/WorkedExampleObjectFactory';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConceptCardBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
 
-      $http.get(conceptCardDataUrl).then(function(response) {
-        var conceptCards = angular.copy(response.data.concept_card_dicts);
+  // Maps previously loaded concept cards to their IDs.
+  private _conceptCardCache = [];
+  private conceptCardObjectFactory = null;
+
+  private _fetchConceptCards(
+      skillIds: Array<string>,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: string) => void): void {
+    var conceptCardDataUrl = this.urlInterpolation.interpolateUrl(
+      SkillDomainConstants.CONCEPT_CARD_DATA_URL_TEMPLATE, {
+        comma_separated_skill_ids: skillIds.join(',')
+      });
+      
+    var conceptCardObjects = [];
+    var conceptCardObjectFactory = new ConceptCardObjectFactory(
```

Comment:
Do not use this to declare conceptCardObjectFactory. Please inject this into the constructor similar to what you've done for UrlInterpolation.

Reply:
Fixed!